test/test_nn.py

```diff
@@ -18948,7 +18957,8 @@ def expected_output(dim):
         output.backward(grad_output, retain_graph=True)
         expected_grad = expected_grad(num_dim)
         # TODO(#38095): Replace assertEqualIgnoreType. See issue #38095
-        self.assertEqualIgnoreType(input_var.grad.data, expected_grad.view_as(input))
+        # self.assertEqualIgnoreType(input_var.grad.data, expected_grad.view_as(input))
+        self.assertEqual(input_var.grad.data.to(dtype=expected_grad.view_as(input).dtype), expected_grad.view_as(input))
```

Comment:
From the code, it looks like `expected_grad.view_as(input).dtype` is always just `torch.double`, use that. And better to modify `expected_grad` dtype, not actual dtype.